### DISCUSSION TITLE
Alterar o Texto HTML Usando JavaScript

### DISCUSSION THREAD

**Discussion by brozmado** [frontendbr/forum#1843] [Perguntas e Respostas]
Olá, estou tentando alterar um texto (Televendas) Para (Contato) incorporado no HTML com DOM Javascript mas sem sucesso!! segue abaixo a estância do que eu fiz:

```
if(document.querySelector(".header-links .container div.col-sm-6 col-xs-12: a")){
    //ALTERA TEXTO TELEVENDAS PARA SIA
    document.querySelector(".header-links .container div.col-sm-6 col-xs-12: a").innerText = "CONTATO";
}
```

![codigo](https://user-images.githubusercontent.com/67505980/85921862-bc5a8000-b855-11ea-850c-65f7214e15c3.png)

Alguém pode me dá uma luz..

**Comment by lucasferreiralimax** [comment]
Olá tudo bem?
Você poderia reduzir esse seletor e também fazer a seleção apenas uma vez

```javascript
let linkContato = document.querySelector("#template_foneTelevendas a");

if(linkContato){
  linkContato.innerText = "Contato:";
}
```

Exemplo rodando online - https://jsfiddle.net/lucaslimax/hf8yaojs

E o erro do seu seletor talvez seja a classe de coluna sem o ponto e os dois pontos após a mesma classe.

Espero ter ajudado abraços :+1:

**Comment by felquis** [comment]
- Use o JavaScript no final do arquivo HTML conheça o [DOMContentLoaded](https://developer.mozilla.org/pt-BR/docs/Web/Events/DOMContentLoaded)
- Certifique-se de que está encontrando o elemento no DOM com [console.dir](https://developer.mozilla.org/pt-BR/docs/Web/API/Console/dir)
- Sempre uso [textContent](https://developer.mozilla.org/pt-BR/docs/Web/API/Node/textContent) para alterar textos.
